### PR TITLE
[5.7] Added support to change session id size

### PR DESCRIPTION
**Pull request by wallaceandrade**
There are cases where you need to increase or decrease the size of the session id. Support has been added so this can be customized in config / session.php by setting the id_size key.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Why do you need to change it?

**Comment by wallaceandrade** [comment]
Estamos migrando um site com cerca de 90 mil usuários. O ID de sessão neste site tem 32 caracteres. Se simplesmente migrarmos sem definirmos o ID de sessão do laravel para 32 caracteres em vez de 40, nenhum id de sessão do site antigo será válido e todos os usuários serão deslogados, o que gerará uma grande demanda. Imagino que esse tamanho do ID possa ser configurado via ENV.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Session/Store.php [hunk 1]
```diff
@@ -636,6 +637,16 @@ public function getHandler()
...
     }
 
+    /**
+     * Return session id required size.
+     * 
+     * @return int
+     */
+    public function getIdSize()
+    {
+        return (isset($_ENV['SESSION_ID_SIZE']) && is_numeric($_ENV['SESSION_ID_SIZE'])) ? $_ENV['SESSION_ID_SIZE'] : 40;
```
[the remark this answers is not in the document]
Comment (wallaceandrade) [reply src/Illuminate/Session/Store.php]: Should I use the env () helper instead? Or the helper config ()? what would be the way?

[the remark this answers is not in the document]
Comment (sisve) [reply src/Illuminate/Session/Store.php]: Use the config() helper. This is related to cached configuration files and how the use of config:cache will not use the .env file.

[the remark this answers is not in the document]
Comment (X-Coder264) [reply src/Illuminate/Session/Store.php]: We can't use the `config` helper here as `illuminate/session` doesn't require the entire framework (e.g. the `Foundation` namespace) to be there in order to work.

What can be done is that the `Store` receives another parameter through its constructor (with the default value of 40) and then in the `SessionManager` where the `Store` is actually constructed the value from the config can be passed `(buildSession` and `buildEncryptedSession` methods).

**Pull request closed without merging**